src/app/shared/table/table.component.ts

```diff
@@ -134,6 +136,29 @@ export class TableComponent implements OnInit, AfterViewInit, OnDestroy {
     this.refresh();
   }
 
+  public dateRangeChanged(filterDef: TableFilterDef, event: any){
+    const currentValue = filterDef.currentValue;
+    if(currentValue?.startDate !== event.startDate || currentValue?.endDate !== event.endDate){
+      filterDef.currentValue = {
+        startDate: event?.startDate.toDate(),
+        endDate: event?.endDate.toDate()
+      };
+      this.filterChanged(filterDef);
+    }
+  }
+
+  public createQuickAccessDateRanges(rangeObjects: {key: string; label: string; startValue: Date; endValue: Date}[]){
+    const dateRanges: any = {};
+    for (const range of rangeObjects) {
+      dateRanges[this.translateService.instant(range.label)] = [range.startValue, range.endValue];
+    }
+    return dateRanges;
+  }
+
+  public openDateRange() {
+    this.picker.open();
```

Comment:
Can't this be the default behavior without having to provide this code?

Reply:
Removed the calendar icon. The [documentation](https://github.com/fetrarij/ngx-daterangepicker-material#open-datepicker-from-outside) doesn't provide an option to access the injected element within the DOM without this piece of code. We need to find a way to accomplish this.